copier/main.py

```diff
@@ -241,6 +243,12 @@ def _check_unsafe(self, mode: Literal["copy", "update"]) -> None:
         if features:
             raise UnsafeTemplateError(sorted(features))
 
+    def _print_message(self, message: str) -> None:
+        if message and not self.quiet:
+            Console(stderr=True).print(
+                Markdown(self._render_string(message), hyperlinks=True)
```

Comment:
The call to `self._render_string(...)` starts the questionnaire, so the `message_before_copy` is rendered _after_ answering the questions. :thinking:

Reply:
It's a side effect of using an [impure property for prompting](https://github.com/copier-org/copier/blob/1b351bc8560d19d70e927cdb2fafb4f7ce41092d/copier/main.py#L411-L463) instead of a method that records the answers in the state of the `Worker` class.

Reply:
I've submitted #1198 to trigger the questionnaire imperatively and, thus, get explicit control, so that the render context can be accessed prior to the questionnaire, of course without containing any answers then.